test/libsolidity/syntaxTests/inlineAssembly/external_identifier_access_shadowing.sol

```diff
@@ -0,0 +1,12 @@
+contract C {
+    function f() public returns (uint x) {
+        assembly {
+            function g() -> x {
+                x := 42
+            }
+            x := g()
+        }
+    }
+}
+// ----
+// DeclarationError 6578: (123-124): Cannot access local Solidity variables from inside an inline assembly function.
```

Comment:
Yeah, that's of course a bit weird... but if we make the ``-> x`` a proper shadowing error later on, I guess it's fine for now.

Reply:
By the way, this is not a new error.